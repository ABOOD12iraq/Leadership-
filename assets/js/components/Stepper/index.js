/**
 * Stepper component.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * WordPress dependencies
 */
import { Children, cloneElement } from '@wordpress/element';
import { __, sprintf } from '@wordpress/i18n';

import Tick from '../../../svg/icons/tick.svg';
import { STEP_STATUS } from './constants';

// English-language labels for the step statuses.
const STEP_STATUS_LABEL = {
	[ STEP_STATUS.COMPLETED ]: 'completed',
	[ STEP_STATUS.ACTIVE ]: 'active',
	[ STEP_STATUS.UPCOMING ]: 'upcoming',
};

export default function Stepper( { children, activeStep } ) {
	function getStepStatus( index = -1 ) {
		if ( index < activeStep ) {
			return STEP_STATUS.COMPLETED;
		}

		if ( index === activeStep ) {
			return STEP_STATUS.ACTIVE;
		}

		return STEP_STATUS.UPCOMING;
	}

	const childCount = Children.count( children );

	return (
		<ol className="googlesitekit-stepper">
			{ Children.map( children, ( child, childIndex ) => {
				const stepStatus = getStepStatus( childIndex, activeStep );

				return (
					<li className="googlesitekit-stepper__step">
						<div className="googlesitekit-stepper__step-progress">
							<span
								className={ `googlesitekit-stepper__step-number googlesitekit-stepper__step-number--${ stepStatus }` }
								title={ sprintf(
									/* translators: 1: The number of the current step. 2: The total number of steps. 3: The status of the step ("completed", "active" or "upcoming"). */
									__(
										'Step %1$s of %2$s (%3$s).',
										'google-site-kit'
									),
									childIndex + 1,
									childCount,
									STEP_STATUS_LABEL[ stepStatus ]
								) }
							>
								{ stepStatus === STEP_STATUS.COMPLETED ? (
									<Tick />
								) : (
									childIndex + 1
								) }
							</span>
							{ childIndex < childCount - 1 && (
								<div className="googlesitekit-stepper__step-progress-line"></div>
							) }
						</div>
						{ cloneElement( child, { stepStatus } ) }
					</li>
				);
			} ) }
		</ol>
	);
}

Stepper.propTypes = {
	children: PropTypes.node.isRequired,
	// The index of the active step. If omitted or negative, all steps are in the upcoming state. If greater than the number of steps - 1, all steps are complete.
	activeStep: PropTypes.number,
};
