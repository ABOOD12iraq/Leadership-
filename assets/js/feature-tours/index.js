/**
 * Feature Tours.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { VIEW_CONTEXT_DASHBOARD } from '../googlesitekit/constants';
import { CORE_MODULES } from '../googlesitekit/modules/datastore/constants';

const allTrafficWidget = {
	slug: 'allTrafficWidget',
	contexts: [ VIEW_CONTEXT_DASHBOARD ],
	version: '1.25.0',
	checkRequirements: async ( registry ) => {
		// Here we need to wait for the underlying selector to be resolved before selecting `isModuleConnected`.
		await registry.__experimentalResolveSelect( CORE_MODULES ).getModules();
		if ( ! registry.select( CORE_MODULES ).isModuleConnected( 'analytics' ) ) {
			return false;
		}
		return true;
	},
	steps: [
		{
			target: '.googlesitekit-widget--analyticsAllTraffic__user-count-chart',
			title: __( "It's now easier to see your site's traffic at a glance", 'google-site-kit' ),
			content: __( 'Check the trend graph to see how your traffic changed over time', 'google-site-kit' ),
		},
		{
			target: '.googlesitekit-widget--analyticsAllTraffic__dimensions-chart',
			title: __( 'See where your visitors come from', 'google-site-kit' ),
			content: __( 'Click on the chart slices to see how each segment has changed over time', 'google-site-kit' ),
		},
		{
			target: '.googlesitekit-header__date-range-selector-menu',
			title: __( 'Check how your traffic changed since you last looked', 'google-site-kit' ),
			content: __( 'Select a time frame to see the comparison with the previous time period', 'google-site-kit' ),
		},
	],
};

// Ordered tours.
export default [
	allTrafficWidget,
];
