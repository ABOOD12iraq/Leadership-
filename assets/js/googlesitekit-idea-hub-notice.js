/**
 * Idea Hub Notice (for Block Editor) entrypoint.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __, sprintf } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { getItem, setItem } from './googlesitekit/api/cache';

const shownNotices = [];

const editorNoticeKey = ( postID ) => {
	return `modules::idea-hub::dismissed-editor-notice-${ postID }`;
};

const loadIdeaHubNotices = async ( _global = global ) => {
	const { wp } = _global;

	const hasNotice = ( postID ) => {
		if ( wp.data.select( 'core/notices' ).getNotices() === undefined ) {
			return undefined;
		}

		return wp.data.select( 'core/notices' ).getNotices().some( ( notice ) => {
			return notice.id === editorNoticeKey( postID );
		} );
	};

	const listener = async () => {
		// eslint-disable-next-line sitekit/acronym-case
		const postID = wp.data.select( 'core/editor' ).getCurrentPostId();

		if ( ! postID ) {
			return;
		}

		const { cacheHit } = await getItem( editorNoticeKey( postID ) );

		if ( hasNotice( postID ) === false ) {
			// We've already shown this notice, so when it's hidden, mark it as shown
			// so it doesn't appear again.
			if ( shownNotices.includes( editorNoticeKey( postID ) ) ) {
				setItem(
					editorNoticeKey( postID ),
					postMeta.googlesitekitpersistent_idea_text,
					{ ttl: 108000 } // Don't show this notice for another 90 days.
				);
				unsubscribeFromListener();
				return;
			}

			// We've already shown this notice on a previous visit to this page in
			// the editor, so don't show it again.
			if ( cacheHit ) {
				unsubscribeFromListener();
				return;
			}

			// We haven't shown any notice for this post before, so let's check for
			// Idea Hub postmeta.
			const postMeta = wp.data.select( 'core/editor' ).getEditedPostAttribute( 'meta' );

			// eslint-disable-next-line camelcase
			if ( postMeta?.googlesitekitpersistent_idea_text ) {
				wp.data.dispatch( 'core/notices' ).createInfoNotice(
					/* translators: %s: Idea post name */
					sprintf( __( 'This post was created from an idea you picked in Site Kitâ€™s Idea Hub: %s', 'google-site-kit' ), postMeta.googlesitekitpersistent_idea_text ),
					{ id: editorNoticeKey( postID ) }
				);

				// Mark the notice as shown locally, so if it is no longer marked as
				// visible, we known it was dismissed (rather than having yet to be
				// displayed).
				shownNotices.push( editorNoticeKey( postID ) );
			}
		}
	};

	const unsubscribeFromListener = wp.data.subscribe( listener );
};

loadIdeaHubNotices();
