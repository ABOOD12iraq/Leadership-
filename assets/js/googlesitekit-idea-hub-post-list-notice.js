/**
 * Idea Hub Post List notice.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
// import './modules';

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
// import { render } from '@wordpress/element';

/**
 * Internal dependencies
 */
import Data from 'googlesitekit-data';
const { dispatch } = Data;
import { CORE_USER } from './googlesitekit/datastore/user/constants';

const WEEK_IN_SECONDS = 3600 * 24 * 7;

const dismissNotice = ( slug, expirationTime ) => dispatch( CORE_USER ).dismissItem( slug, expirationTime );

domReady( () => {
	const newNotice = document.getElementById( 'googlesitekit-notice-new' );
	const savedNotice = document.getElementById( 'googlesitekit-notice-saved' );
	let type, notice;
	if ( newNotice ) {
		type = 'new';
		notice = newNotice;
	} else if ( savedNotice ) {
		type = 'saved';
		notice = savedNotice;
	} else {
		return;
	}

	// Button pops up only after the timeout passes
	setTimeout( () => {
		const button = notice.getElementsByClassName( 'notice-dismiss' )[ 0 ];
		const slug = type === 'new' ? 'new-ideas' : 'saved-ideas';
		const expirationType = type === 'new' ? 0 : WEEK_IN_SECONDS;
		button.addEventListener( 'click', () => dismissNotice( slug, expirationType ) );
	}, 1 );
} );
