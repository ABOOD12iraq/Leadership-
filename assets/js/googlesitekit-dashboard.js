/**
 * Dashboard component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
import { render } from '@wordpress/element';

/**
 * Internal dependencies
 */
import API from 'googlesitekit-api';
import { clearWebStorage } from './util';
import Root from './components/Root';
import './components/legacy-notifications';
import {
	VIEW_CONTEXT_DASHBOARD,
	VIEW_CONTEXT_DASHBOARD_VIEW_ONLY,
	VIEW_CONTEXT_MODULE_SETUP,
} from './googlesitekit/constants';
import DashboardEntryPoint from './components/DashboardEntryPoint';

// Initialize the app once the DOM is ready.
domReady( async () => {
	if ( global._googlesitekitLegacyData.admin.resetSession ) {
		clearWebStorage();
	}

	const renderTarget = document.getElementById(
		'js-googlesitekit-dashboard'
	);

	if ( renderTarget ) {
		// Make a separate API request since we cannot use selectors
		// outside the Root component.
		const { authenticated: isAuthenticated } = await API.get(
			'core',
			'user',
			'authentication',
			{},
			{ useCache: false }
		);

		const { setupModuleSlug } = renderTarget.dataset;
		let viewContext = VIEW_CONTEXT_MODULE_SETUP;
		if ( ! setupModuleSlug ) {
			viewContext = isAuthenticated
				? VIEW_CONTEXT_DASHBOARD
				: VIEW_CONTEXT_DASHBOARD_VIEW_ONLY;
		}
		render(
			<Root viewContext={ viewContext }>
				<DashboardEntryPoint setupModuleSlug={ setupModuleSlug } />
			</Root>,
			renderTarget
		);
	}
} );
