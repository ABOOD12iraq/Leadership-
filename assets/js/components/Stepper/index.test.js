/**
 * Stepper component tests.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import { render } from '../../../../tests/js/test-utils';
import Step from './Step';
import Stepper from '.';

describe( 'Stepper', () => {
	it( 'should render the active step content', () => {
		const { container, rerender } = render(
			<Stepper activeStep={ 0 }>
				<Step title="Step 1">This is step 1.</Step>
				<Step title="Step 2">This is step 2.</Step>
			</Stepper>
		);

		expect( container ).toMatchSnapshot();

		expect(
			container.querySelector(
				'.googlesitekit-stepper__step:nth-child(1) .googlesitekit-step__content'
			)
		).toBeInTheDocument();

		expect(
			container.querySelector(
				'.googlesitekit-stepper__step:nth-child(2) .googlesitekit-step__content'
			)
		).not.toBeInTheDocument();

		rerender(
			<Stepper activeStep={ 1 }>
				<Step title="Step 1">This is step 1.</Step>
				<Step title="Step 2">This is step 2.</Step>
			</Stepper>
		);

		expect( container ).toMatchSnapshot();

		expect(
			container.querySelector(
				'.googlesitekit-stepper__step:nth-child(1) .googlesitekit-step__content'
			)
		).not.toBeInTheDocument();

		expect(
			container.querySelector(
				'.googlesitekit-stepper__step:nth-child(2) .googlesitekit-step__content'
			)
		).toBeInTheDocument();
	} );
} );
