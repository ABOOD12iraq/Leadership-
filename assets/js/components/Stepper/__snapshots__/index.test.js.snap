// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper should render the active step content 1`] = `
<div>
  <ol
    class="googlesitekit-stepper"
  >
    <li
      class="googlesitekit-stepper__step"
    >
      <div
        class="googlesitekit-stepper__step-progress"
      >
        <span
          class="googlesitekit-stepper__step-number googlesitekit-stepper__step-number--active"
          title="Step 1 of 2 (active)."
        >
          1
        </span>
        <div
          class="googlesitekit-stepper__step-progress-line"
        />
      </div>
      <div
        class="googlesitekit-step"
      >
        <h2
          class="googlesitekit-step__title googlesitekit-step__title--active"
        >
          Step 1
        </h2>
        <div
          class="googlesitekit-step__content"
        >
          This is step 1.
        </div>
      </div>
    </li>
    <li
      class="googlesitekit-stepper__step"
    >
      <div
        class="googlesitekit-stepper__step-progress"
      >
        <span
          class="googlesitekit-stepper__step-number googlesitekit-stepper__step-number--upcoming"
          title="Step 2 of 2 (upcoming)."
        >
          2
        </span>
      </div>
      <div
        class="googlesitekit-step"
      >
        <h2
          class="googlesitekit-step__title googlesitekit-step__title--upcoming"
        >
          Step 2
        </h2>
      </div>
    </li>
  </ol>
</div>
`;

exports[`Stepper should render the active step content 2`] = `
<div>
  <ol
    class="googlesitekit-stepper"
  >
    <li
      class="googlesitekit-stepper__step"
    >
      <div
        class="googlesitekit-stepper__step-progress"
      >
        <span
          class="googlesitekit-stepper__step-number googlesitekit-stepper__step-number--completed"
          title="Step 1 of 2 (completed)."
        >
          <svg />
        </span>
        <div
          class="googlesitekit-stepper__step-progress-line"
        />
      </div>
      <div
        class="googlesitekit-step"
      >
        <h2
          class="googlesitekit-step__title googlesitekit-step__title--completed"
        >
          Step 1
        </h2>
      </div>
    </li>
    <li
      class="googlesitekit-stepper__step"
    >
      <div
        class="googlesitekit-stepper__step-progress"
      >
        <span
          class="googlesitekit-stepper__step-number googlesitekit-stepper__step-number--active"
          title="Step 2 of 2 (active)."
        >
          2
        </span>
      </div>
      <div
        class="googlesitekit-step"
      >
        <h2
          class="googlesitekit-step__title googlesitekit-step__title--active"
        >
          Step 2
        </h2>
        <div
          class="googlesitekit-step__content"
        >
          This is step 2.
        </div>
      </div>
    </li>
  </ol>
</div>
`;
